force_wfmanager/central_pane/setup_pane.py

```diff
@@ -1,25 +1,118 @@
+    #: this is set to 'None' (with type Unicode!)
+    selected_factory_name = Unicode('Workflow')
+
+    #: The appropriate function to add a new entity of the selected factory
+    #: group to the workflow tree. For example, if the 'DataSources' group
+    #: is selected, this function would be new_data_source().
+    add_new_entity_function = Callable()
+
+    #: Function to remove the currently selected instance
+    remove_entity_function = Callable()
+
+    #: A NewEntityModal object displaying the factories of the currently
+    #: selected group
+    current_modal = Instance(NewEntityModal)
+
+    #: A Button which calls add_new_entity_function when pressed
+    add_new_entity = Button()
+
+    #: A Button which calls remove_entity_function when pressed
+    remove_entity = Button()
+
+    #: Switch to disable adding an entity if all required properties are
+    #: not set
+    enable_add_button = Property(Bool, depends_on='current_modal,'
+                                                  'current_modal.model')
+
+    #: The view when editing an existing instance within the workflow tree
+    traits_view = View(
+        HGroup(
+                # Instance View
+                VGroup(
+                    UItem("selected_mv", editor=InstanceEditor(),
+                          style="custom",
+                          visible_when="selected_mv_editable",
+                          ),
+                    UItem("selected_model", editor=InstanceEditor(),
+                          style="custom",
+                          visible_when="selected_model is not None"),
+                    HGroup(
+                        UItem('remove_entity', label='Delete'),
+                    ),
+                    label="Model Details",
+                    visible_when="selected_factory_name=='None'",
+                    show_border=True,
+                    ),
+                # Factory View
+                VGroup(
+                    HGroup(
+                        UItem("current_modal", editor=InstanceEditor(),
+                              style="custom",
+                              visible_when="current_modal is not None",
+                              width=825),
+                        springy=True,
+
+                    ),
+                    HGroup(
+                        UItem('add_new_entity', label='Add',
+                              enabled_when='enable_add_button'),
+                        UItem('remove_entity', label='Delete Layer',
+                              visible_when="selected_factory_name == "
+                                           "'DataSources'"),
+                    ),
+                    label="New Model Details",
+                    visible_when="selected_factory_name != 'None'",
+                    show_border=True,
+
+                ),
 
-    def __init__(self, *args, **kwargs):
-        super(SetupPane, self).__init__(*args, **kwargs)
+                # TODO:
```

Comment:
I sometime uncomment this for debug things still, so it would be helpful to keep it around for now. As it says, it will be replaced in the course of the UI redesign